Android deletemedia fix

shahzaib14944 opened the pull request:
<!-- Thanks for submitting a pull request! We appreciate you spending the time to work on these changes. Please follow the template so that the reviewers can easily understand what the code changes affect -->

# Summary

<!--
Explain the **motivation** for making this change: here are some points to help you:
* What issues does the pull request solve? Please tag them so that they will get automatically closed once the PR is merged
* What is the feature? (if applicable)
* How did you implement the solution?
* What areas of the library does it impact?
-->

### Pull Request Summary

This pull request addresses the outstanding issue of media file deletion in Android, specifically resolving [Issue #639: Android: deletePhotos not working with content:// assets](https://github.com/react-native-cameraroll/react-native-cameraroll/issues/639).

### Key Changes

1. **Deletion Method Update**:
   - The `ContentResolver.delete()` functionality has been replaced with `MediaStore.createDeleteRequest()`. This change enhances compatibility with scoped storage requirements introduced in Android 10 and above. 

2. **User Permission Handling**:
   - For media files not created by the app, explicit user permission will now be required for deletion.

3. **Updated URI Format**:
   - Media URIs will now conform to the following formats:
     - Video Example: `content://media/external/video/media/1000005715`
     - Image Example: `content://media/external/images/media/1000005714`

### Affected Files
- **CameraRollModule.java**

This update aims to improve the robustness of the CameraRoll library and enhance user experience by adhering to modern Android storage practices.


## Test Plan
```
  const deletePhotos = async () => {
    if (selectedUris.length === 0) {
      Alert.alert("No photos selected", "Please select photos to delete.");
      return;
    }
    
    try {
      await CameraRoll.deletePhotos(selectedUris);
      Alert.alert("Success", "Photos deleted successfully.");
      // Refresh the photo list
      setSelectedUris([]);
      fetchPhotos();
    } catch (error) {
      console.error("Error deleting photos: ", error);
      Alert.alert("Error", String(error));
    }
  };
```
Result: (Native Android Popup):
![app](https://github.com/user-attachments/assets/8bcb0362-c9ed-4770-a6a9-2c16c35aa941)


<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI. -->

### What's required for testing (prerequisites)?
Go to example folder, and run "npm run android" and you're good to go.

### What are the steps to reproduce (after prerequisites)?


## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ❌     |
| Android |    ✅   |

## Checklist

<!-- Check completed item, when applicable, via: [X] -->

- [x] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I updated the typed files (TS and Flow)
- [x] I added a sample use of the API in the example project (`example/App.js`)

Arjan-Zuidema commented:
@shahzaib14944 Thank you for this PR! I see you overhauled the example, which is nice. 
Can you also add an example in the FabricExample? I would like to be able to test this in both versions.

I know; it would be better if both examples used the same code, which is a refactor I would like to do some day 😄

shahzaib14944 commented:
> @shahzaib14944 Thank you for this PR! I see you overhauled the example, which is nice. Can you also add an example in the FabricExample? I would like to be able to test this in both versions.
> 
> I know; it would be better if both examples used the same code, which is a refactor I would like to do some day 😄

Hello @Arjan-Zuidema ,

I’ve added the example to FabricExample and updated App.js (previously App.tsx). Everything is working fine on my end, and I believe this aligns with the requirements.

Please feel free to review and suggest any improvements to make it better!

Arjan-Zuidema commented:
May I ask why you replaced the typescript file with javascript?

shahzaib14944 commented:
> May I ask why you replaced the typescript file with javascript?

I opted to use JavaScript to provide a quicker example, particularly since I was able to reuse code from a previous example.

Arjan-Zuidema commented:
Sorry for the (very) late reply. But I'd rather use typescript in the examples

shahzaib14944 commented:
> Sorry for the (very) late reply. But I'd rather use typescript in the examples

I've updated the examples; replaced JS with TS, hope you're happy now! :lay

Arjan-Zuidema commented:
@shahzaib14944 Thanks! Will review and merge this week I hope!

bartolkaruza commented:
:tada: This PR is included in version 7.8.4 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@react-native-camera-roll/camera-roll/v/7.8.4)
- [GitHub release](https://github.com/react-native-cameraroll/react-native-cameraroll/releases/tag/v7.8.4)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
